File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) 
    {
        int len=s.length();
        unordered_set<char> visit;
        int ans=0;
        for(int i=0;i<len;i++)
        {
            visit.clear();
            int n=0;
            for(int j=i;j<len;j++)
            {
                if(visit.count(s[j]))
                {
                    break;
                }
                else
                {
                    visit.insert(s[j]);
                    n++;
                }
                ans=max(ans,n);
            }
        }
        return ans;
    }
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int i=0;
        int j=0;
        int max=0;
        HashSet<Character>st=new HashSet<>();
        while(j<s.length())
        {
            if(!st.contains(s.charAt(j)))
            {
                st.add(s.charAt(j));
                j++;
                max=Math.max(max,st.size());
            }
            else
            {
                st.remove(s.charAt(i));
                i++;
            }
        }
        return max;
    }
}